packages/cli-kit/src/private/node/ui/hooks/use-async.ts

```diff
@@ -0,0 +1,26 @@
+import {useApp} from 'ink'
+import {useEffect} from 'react'
+
+interface Options {
+  onResolve?: () => unknown
+  onReject?: (error: Error) => void
+}
+
+export default function useAsync(
+  asyncFunction: () => Promise<unknown>,
+  {onResolve = () => {}, onReject = () => {}}: Options,
```

Comment:
Nitpick: What about aligning the [naming](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#parameters) of the `then` function: `onFulfilled` and `onRejected`?